src/lens/follow.js

```diff
@@ -0,0 +1,83 @@
+import { gql } from '@apollo/client/core';
+import { apolloClient } from '../apollo-client';
+import { signedTypeData, splitSignature, getAddress } from '@/ethers-service';
+import { lensHub } from '@/lens/utils/lens-hub';
+
+const CREATE_FOLLOW_TYPED_DATA = `
+  mutation($request: FollowRequest!) { 
+    createFollowTypedData(request: $request) {
+      id
+      expiresAt
+      typedData {
+        domain {
+          name
+          chainId
+          version
+          verifyingContract
+        }
+        types {
+          FollowWithSig {
+            name
+            type
+          }
+        }
+        value {
+          nonce
+          deadline
+          profileIds
+          datas
+        }
+      }
+    }
+ }
+`;
+
+// TODO sort typed!
+const createFollowTypedData = (followRequestInfo) => {
+  return apolloClient.mutate({
+    mutation: gql(CREATE_FOLLOW_TYPED_DATA),
+    variables: {
+      request: {
+        follow: followRequestInfo,
+      },
+    },
+  });
+};
+
+export const follow = async (profileId = '0x12') => {
```

Comment:
Why did you put a default value for profileId ?

Also, same as before, I added this :
```javascript
if (!profileId) {
    throw new Error('profileId is undefined');
}
```